src/main/java/com/google/devtools/build/lib/remote/AbstractActionInputPrefetcher.java

```diff
@@ -569,26 +570,33 @@ private Completable downloadFileNoCheckRx(
     }
 
     Path finalPath = path;
+    PathFragment execPath = finalPath.relativeTo(execRoot);
```

Comment:
Does this handle symlinks correctly? We need the path of the artifact, not the download exec path.

Reply:
Symlink is handled by the call site as the form of `downloadFileNoCheckRx().onErrorResumeNext(...)`.